### DISCUSSION TITLE
Pycharm keeps crashing every time I run reloadium

### DISCUSSION THREAD

**Discussion by cryptopotluck** [reloadware/reloadium#70] [Q&A] [answered]
Hey, really looks like a useful plugin. However, every time I try and run reloadium it crashes pycharm. For context all I did was install the plugin via pycharms plugin store and when I click the run this is what happens:
<img width="1088" alt="Screen Shot 2022-11-21 at 8 34 43 PM" src="https://user-images.githubusercontent.com/40912998/203207006-e8ecada8-7f95-43ee-8210-7a96f93685e2.png">

Also included the crash log, maybe its a bug or an issue with my machine specifically but figured I'd share it with you:
Process:               Python [11058]
Path:                  /Library/Frameworks/Python.framework/Versions/3.7/Resources/Python.app/Contents/MacOS/Python
Identifier:            Python
Version:               3.7.0 (3.7.0)
Code Type:             X86-64 (Native)
Parent Process:        pycharm [10769]
Responsible:           pycharm [10769]
User ID:               501

Date/Time:             2022-11-21 20:35:17.312 -0600
OS Version:            Mac OS X 10.15.6 (19G73)
Report Version:        12
Bridge OS Version:     3.0 (14Y908)
Anonymous UUID:        69059D8B-865F-4BD7-FD8C-A59E1186BF83

Sleep/Wake UUID:       2FA1EDA0-C3E7-4532-BF2A-0829AF1802B0

Time Awake Since Boot: 31000 seconds
Time Since Wake:       25000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000100000059
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [11058]

VM Regions Near 0x100000059:
--> 
    __TEXT                 000000010f785000-000000010f786000 [    4K] r-x/rwx SM=COW  /Library/Frameworks/Python.framework/Versions/3.7/Resources/Python.app/Contents/MacOS/Python

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   corium.cpython-37m-darwin.so  	0x000000011008a277 Nuitka_CheckFunctionResult + 29
1   corium.cpython-37m-darwin.so  	0x000000011008425d CALL_FUNCTION_WITH_ARGS5 + 1346
2   corium.cpython-37m-darwin.so  	0x0000000110083d00 IMPORT_MODULE5 + 85
3   corium.cpython-37m-darwin.so  	0x00000001100d5da7 patchInspectModule + 56
4   corium.cpython-37m-darwin.so  	0x000000010fad814c modulecode_corium + 112
5   org.python.python             	0x000000010f88bc86 _PyImport_LoadDynamicModuleWithSpec + 662
6   org.python.python             	0x000000010f88b6a8 _imp_create_dynamic + 296
7   org.python.python             	0x000000010f7ac398 _PyMethodDef_RawFastCallDict + 248
8   org.python.python             	0x000000010f7ac081 PyCFunction_Call + 209
9   org.python.python             	0x000000010f8689f2 _PyEval_EvalFrameDefault + 26322
10  org.python.python             	0x000000010f86c096 _PyEval_EvalCodeWithName + 2422
11  org.python.python             	0x000000010f7abd81 _PyFunction_FastCallKeywords + 257
12  org.python.python             	0x000000010f86b5bb call_function + 795
13  org.python.python             	0x000000010f868560 _PyEval_EvalFrameDefault + 25152
14  org.python.python             	0x000000010f7ac1f0 function_code_fastcall + 128
15  org.python.python             	0x000000010f86b5bb call_function + 795
16  org.python.python             	0x000000010f868546 _PyEval_EvalFrameDefault + 25126
17  org.python.python             	0x000000010f7ac1f0 function_code_fastcall + 128
18  org.python.python             	0x000000010f86b5bb call_function + 795
19  org.python.python             	0x000000010f868601 _PyEval_EvalFrameDefault + 25313
20  org.python.python             	0x000000010f7ac1f0 function_code_fastcall + 128
21  org.python.python             	0x000000010f86b5bb call_function + 795
22  org.python.python             	0x000000010f868601 _PyEval_EvalFrameDefault + 25313
23  org.python.python             	0x000000010f7ac1f0 function_code_fastcall + 128
24  org.python.python             	0x000000010f86b5bb call_function + 795
25  org.python.python             	0x000000010f868601 _PyEval_EvalFrameDefault + 25313
26  org.python.python             	0x000000010f7ac1f0 function_code_fastcall + 128
27  org.python.python             	0x000000010f7ab7e4 _PyFunction_FastCallDict + 148
28  org.python.python             	0x000000010f7ad8bc object_vacall + 316
29  org.python.python             	0x000000010f7ad9b1 _PyObject_CallMethodIdObjArgs + 177
30  org.python.python             	0x000000010f88a5d9 PyImport_ImportModuleLevelObject + 1449
31  org.python.python             	0x000000010f866ed5 _PyEval_EvalFrameDefault + 19381
32  org.python.python             	0x000000010f7ac1f0 function_code_fastcall + 128
33  org.python.python             	0x000000010f86b5bb call_function + 795
34  org.python.python             	0x000000010f868601 _PyEval_EvalFrameDefault + 25313
35  org.python.python             	0x000000010f86c096 _PyEval_EvalCodeWithName + 2422
36  org.python.python             	0x000000010f862244 PyEval_EvalCode + 100
37  org.python.python             	0x000000010f85f85d builtin_exec + 557
38  org.python.python             	0x000000010f7ac398 _PyMethodDef_RawFastCallDict + 248
39  org.python.python             	0x000000010f7ac081 PyCFunction_Call + 209
40  org.python.python             	0x000000010f8689f2 _PyEval_EvalFrameDefault + 26322
41  org.python.python             	0x000000010f86c096 _PyEval_EvalCodeWithName + 2422
42  org.python.python             	0x000000010f7abd81 _PyFunction_FastCallKeywords + 257
43  org.python.python             	0x000000010f86b5bb call_function + 795
44  org.python.python             	0x000000010f868560 _PyEval_EvalFrameDefault + 25152
45  org.python.python             	0x000000010f7ac1f0 function_code_fastcall + 128
46  org.python.python             	0x000000010f86b5bb call_function + 795
47  org.python.python             	0x000000010f868546 _PyEval_EvalFrameDefault + 25126
48  org.python.python             	0x000000010f7ac1f0 function_code_fastcall + 128
49  org.python.python             	0x000000010f86b5bb call_function + 795
50  org.python.python             	0x000000010f868601 _PyEval_EvalFrameDefault + 25313
51  org.python.python             	0x000000010f7ac1f0 function_code_fastcall + 128
52  org.python.python             	0x000000010f86b5bb call_function + 795
53  org.python.python             	0x000000010f868601 _PyEval_EvalFrameDefault + 25313
54  org.python.python             	0x000000010f7ac1f0 function_code_fastcall + 128
55  org.python.python             	0x000000010f7ab7e4 _PyFunction_FastCallDict + 148
56  org.python.python             	0x000000010f7ad8bc object_vacall + 316
57  org.python.python             	0x000000010f7ad9b1 _PyObject_CallMethodIdObjArgs + 177
58  org.python.python             	0x000000010f88a5d9 PyImport_ImportModuleLevelObject + 1449
59  org.python.python             	0x000000010f85ed47 builtin___import__ + 135
60  org.python.python             	0x000000010f7ac94d _PyMethodDef_RawFastCallKeywords + 685
61  org.python.python             	0x000000010f7abdba _PyCFunction_FastCallKeywords + 42
62  org.python.python             	0x000000010f86b5ae call_function + 782
63  org.python.python             	0x000000010f868601 _PyEval_EvalFrameDefault + 25313
64  org.python.python             	0x000000010f86c096 _PyEval_EvalCodeWithName + 2422
65  org.python.python             	0x000000010f7abd81 _PyFunction_FastCallKeywords + 257
66  org.python.python             	0x000000010f86b5bb call_function + 795
67  org.python.python             	0x000000010f868601 _PyEval_EvalFrameDefault + 25313
68  org.python.python             	0x000000010f86c096 _PyEval_EvalCodeWithName + 2422
69  org.python.python             	0x000000010f7abd81 _PyFunction_FastCallKeywords + 257
70  org.python.python             	0x000000010f86b5bb call_function + 795
71  org.python.python             	0x000000010f868601 _PyEval_EvalFrameDefault + 25313
72  org.python.python             	0x000000010f86c096 _PyEval_EvalCodeWithName + 2422
73  org.python.python             	0x000000010f7ab95b _PyFunction_FastCallDict + 523
74  org.python.python             	0x000000010f8bce63 pymain_run_module + 147
75  org.python.python             	0x000000010f8bc1a9 pymain_main + 6105
76  org.python.python             	0x000000010f8bcda1 _Py_UnixMain + 129
77  libdyld.dylib                 	0x00007fff68fa7cc9 start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000100000001  rbx: 0x00007f9bcef42db8  rcx: 0x0000000000000002  rdx: 0x000000010f9ca210
  rdi: 0x00007f9bcef42db8  rsi: 0x00007f9bcef5ad58  rbp: 0x00007ffee0477120  rsp: 0x00007ffee0477100
   r8: 0x00007f9bcd65a648   r9: 0x0000000000000000  r10: 0x0000000000000001  r11: 0x0000000000000000
  r12: 0x000000010f9e61e0  r13: 0x00007ffee04771b0  r14: 0x00007f9bcef42db8  r15: 0x00007f9bcef1a258
  rip: 0x000000011008a277  rfl: 0x0000000000010246  cr2: 0x0000000100000059
  
Logical CPU:     4
Error Code:      0x00000004 (no mapping for user data read)
Trap Number:     14

Binary Images:
       0x10f785000 -        0x10f785fff +org.python.python (3.7.0 - 3.7.0) <4B030EC4-815E-34B7-90E7-D0720C31E072> /Library/Frameworks/Python.framework/Versions/3.7/Resources/Python.app/Contents/MacOS/Python
       0x10f78b000 -        0x10f961ff7 +org.python.python (3.7.0, [c] 2001-2018 Python Software Foundation. - 3.7.0) <4C64F346-20C8-3CBF-B1BD-134CC385C83F> /Library/Frameworks/Python.framework/Versions/3.7/Python
       0x10fac8000 -        0x10fac9fff +_heapq.cpython-37m-darwin.so (0) <E8B35F18-1B5A-3C9E-B1F4-0BE0432459A2> /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/lib-dynload/_heapq.cpython-37m-darwin.so
       0x10facd000 -        0x1100e0fff +corium.cpython-37m-darwin.so (0) <497B183F-1B6D-3914-9C8B-578C874BC703> /Users/USER/*/corium.cpython-37m-darwin.so
       0x110225000 -        0x110225ff7 +_opcode.cpython-37m-darwin.so (0) <11A650B3-FF7B-3DF1-81E2-A906553221C9> /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/lib-dynload/_opcode.cpython-37m-darwin.so
       0x11cbb1000 -        0x11cc42f47  dyld (750.6) <34A11073-9E4C-38C3-9293-7D566ABAE8B6> /usr/lib/dyld
    0x7fff2ee8f000 -     0x7fff2f30efeb  com.apple.CoreFoundation (6.9 - 1677.104) <C0D70026-EDBE-3CBD-B317-367CF4F1C92F> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
    0x7fff65ab6000 -     0x7fff65ab8ff7  libDiagnosticMessagesClient.dylib (112) <C94F3B7B-1854-38EB-9778-834501C53B3F> /usr/lib/libDiagnosticMessagesClient.dylib
    0x7fff65f8c000 -     0x7fff65f8dfff  libSystem.B.dylib (1281.100.1) <001B3B7F-D02C-31D3-B961-1ED445D5A266> /usr/lib/libSystem.B.dylib
    0x7fff66272000 -     0x7fff662c4fff  libc++.1.dylib (902.1) <59A8239F-C28A-3B59-B8FA-11340DC85EDC> /usr/lib/libc++.1.dylib
    0x7fff662c5000 -     0x7fff662daffb  libc++abi.dylib (902) <E692F14F-C65E-303B-9921-BB7E97D77855> /usr/lib/libc++abi.dylib
    0x7fff66d44000 -     0x7fff66d46fff  libfakelink.dylib (149.1) <36146CB2-E6A5-37BB-9EE8-1B4034D8F3AD> /usr/lib/libfakelink.dylib
    0x7fff66e56000 -     0x7fff670adfff  libicucore.A.dylib (64260.0.1) <8AC2CB07-E7E0-340D-A849-186FA1F27251> /usr/lib/libicucore.A.dylib
    0x7fff67dec000 -     0x7fff67e1ffde  libobjc.A.dylib (787.1) <6DF81160-5E7F-3E31-AA1E-C875E3B98AF6> /usr/lib/libobjc.A.dylib
    0x7fff684c8000 -     0x7fff684daff3  libz.1.dylib (76) <793D9643-CD83-3AAC-8B96-88D548FAB620> /usr/lib/libz.1.dylib
    0x7fff68d89000 -     0x7fff68d8eff3  libcache.dylib (83) <5940876E-AC8A-3BE0-80B3-DE3FB14E257A> /usr/lib/system/libcache.dylib
    0x7fff68d8f000 -     0x7fff68d9afff  libcommonCrypto.dylib (60165.120.1) <C095BD55-1D27-337F-9B02-885E1C7FF87A> /usr/lib/system/libcommonCrypto.dylib
    0x7fff68d9b000 -     0x7fff68da2fff  libcompiler_rt.dylib (101.2) <6E80AC11-A277-31FA-AEEF-E5A528274C77> /usr/lib/system/libcompiler_rt.dylib
    0x7fff68da3000 -     0x7fff68dacff7  libcopyfile.dylib (166.40.1) <EB5E0BC8-873D-3546-A40E-C36DC46FA8F6> /usr/lib/system/libcopyfile.dylib
    0x7fff68dad000 -     0x7fff68e3ffdb  libcorecrypto.dylib (866.140.1) <0B6C52DB-5A50-3FCD-8B5E-C0C2F35857E3> /usr/lib/system/libcorecrypto.dylib
    0x7fff68f4c000 -     0x7fff68f8cff0  libdispatch.dylib (1173.100.2) <EAD535EE-1270-39A9-A254-95CF117FF3B0> /usr/lib/system/libdispatch.dylib
    0x7fff68f8d000 -     0x7fff68fc3fff  libdyld.dylib (750.6) <24C41E8B-6B33-30C7-94C9-02D2BD051D66> /usr/lib/system/libdyld.dylib
    0x7fff68fc4000 -     0x7fff68fc4ffb  libkeymgr.dylib (30) <6F582FDB-EB1A-3ED2-A989-B750643E2647> /usr/lib/system/libkeymgr.dylib
    0x7fff68fd2000 -     0x7fff68fd2ff7  liblaunch.dylib (1738.140.1) <AFBCBDD3-0B55-3ECD-8E04-A73A3A57356B> /usr/lib/system/liblaunch.dylib
    0x7fff68fd3000 -     0x7fff68fd8ff7  libmacho.dylib (959.0.1) <1B0296B5-3FD0-342C-BCC2-9886351A4391> /usr/lib/system/libmacho.dylib
    0x7fff68fd9000 -     0x7fff68fdbff3  libquarantine.dylib (110.40.3) <67FE2676-F9E8-3797-AEE5-F5F9D191CFA3> /usr/lib/system/libquarantine.dylib
    0x7fff68fdc000 -     0x7fff68fddff7  libremovefile.dylib (48) <5CEBDAB2-988A-3B66-87BC-3C45D1C08730> /usr/lib/system/libremovefile.dylib
    0x7fff68fde000 -     0x7fff68ff5ff3  libsystem_asl.dylib (377.60.2) <7A07FF86-658E-35D7-8136-829737E98B7B> /usr/lib/system/libsystem_asl.dylib
    0x7fff68ff6000 -     0x7fff68ff6ff7  libsystem_blocks.dylib (74) <0D53847E-AF5F-3ACF-B51F-A15DEA4DEC58> /usr/lib/system/libsystem_blocks.dylib
    0x7fff68ff7000 -     0x7fff6907efff  libsystem_c.dylib (1353.100.2) <AF7873B0-AC4A-3C67-89BB-B8DA87718DAE> /usr/lib/system/libsystem_c.dylib
    0x7fff6907f000 -     0x7fff69082ffb  libsystem_configuration.dylib (1061.141.1) <0EE84C33-64FD-372B-974A-AF7A136F2068> /usr/lib/system/libsystem_configuration.dylib
    0x7fff69083000 -     0x7fff69086fff  libsystem_coreservices.dylib (114) <08B89E9B-C5B2-3E73-8964-03E58692B21F> /usr/lib/system/libsystem_coreservices.dylib
    0x7fff69087000 -     0x7fff6908ffff  libsystem_darwin.dylib (1353.100.2) <BD5BAD5B-AC3F-371A-B4FC-ADF86D6DCE51> /usr/lib/system/libsystem_darwin.dylib
    0x7fff69090000 -     0x7fff69097fff  libsystem_dnssd.dylib (1096.100.3) <01E0965B-83E5-356C-AC1F-C723F5AAB483> /usr/lib/system/libsystem_dnssd.dylib
    0x7fff69098000 -     0x7fff69099ffb  libsystem_featureflags.dylib (17) <6317641E-B43F-36A3-974E-6073786B94B9> /usr/lib/system/libsystem_featureflags.dylib
    0x7fff6909a000 -     0x7fff690e7ff7  libsystem_info.dylib (538) <4DD813CB-D7B0-3AB5-9054-D2E4FF7A64B9> /usr/lib/system/libsystem_info.dylib
    0x7fff690e8000 -     0x7fff69114ff7  libsystem_kernel.dylib (6153.141.1) <2B6311E6-6240-3EF7-8C87-475B66F7452C> /usr/lib/system/libsystem_kernel.dylib
    0x7fff69115000 -     0x7fff6915cfff  libsystem_m.dylib (3178) <77A9D888-36E9-3968-8103-C82FFFC60B9E> /usr/lib/system/libsystem_m.dylib
    0x7fff6915d000 -     0x7fff69184fff  libsystem_malloc.dylib (283.100.6) <059F4DBD-856A-3A59-8B20-FD4B6918C5F2> /usr/lib/system/libsystem_malloc.dylib
    0x7fff69185000 -     0x7fff69192ffb  libsystem_networkextension.dylib (1095.140.2) <4CCE78C9-1DC4-3375-9828-D5BD739D23F3> /usr/lib/system/libsystem_networkextension.dylib
    0x7fff69193000 -     0x7fff6919cff7  libsystem_notify.dylib (241.100.2) <DEAD75F2-DD34-3E8F-82DE-344625A2C25E> /usr/lib/system/libsystem_notify.dylib
    0x7fff6919d000 -     0x7fff691a5fef  libsystem_platform.dylib (220.100.1) <97E825F6-D823-366C-9FF3-B1C8EA891044> /usr/lib/system/libsystem_platform.dylib
    0x7fff691a6000 -     0x7fff691b0fff  libsystem_pthread.dylib (416.100.3) <80B053AA-B6F7-3B59-BC44-78A5A4F7368F> /usr/lib/system/libsystem_pthread.dylib
    0x7fff691b1000 -     0x7fff691b5ff3  libsystem_sandbox.dylib (1217.140.4) <20BF94E2-DAB8-3EBE-AACE-99DF84C1C391> /usr/lib/system/libsystem_sandbox.dylib
    0x7fff691b6000 -     0x7fff691b8fff  libsystem_secinit.dylib (62.100.2) <28EDEB6F-899B-373D-B761-4C20D39285D9> /usr/lib/system/libsystem_secinit.dylib
    0x7fff691b9000 -     0x7fff691c0ffb  libsystem_symptoms.dylib (1238.120.1) <22800D8C-5C2A-3171-8C29-311BBE67F198> /usr/lib/system/libsystem_symptoms.dylib
    0x7fff691c1000 -     0x7fff691d7ff2  libsystem_trace.dylib (1147.120) <DC00C77E-4709-31F0-B913-CF09559B7A6F> /usr/lib/system/libsystem_trace.dylib
    0x7fff691d9000 -     0x7fff691deff7  libunwind.dylib (35.4) <42B7B509-BAFE-365B-893A-72414C92F5BF> /usr/lib/system/libunwind.dylib
    0x7fff691df000 -     0x7fff69214ffe  libxpc.dylib (1738.140.1) <58E276A9-EE11-3F02-9D3E-5371E604E677> /usr/lib/system/libxpc.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 30030
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=409.8M resident=0K(0%) swapped_out_or_unallocated=409.8M(100%)
Writable regions: Total=92.1M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=92.1M(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Kernel Alloc Once                    8K        1 
MALLOC                            75.5M       38 
MALLOC guard page                   16K        4 
MALLOC_LARGE (reserved)            128K        1         reserved VM address space (unallocated)
STACK GUARD                          4K        1 
Stack                             16.0M        1 
__DATA                            3197K       52 
__DATA_CONST                        20K        1 
__LINKEDIT                       390.3M        8 
__OBJC_RO                         32.3M        1 
__OBJC_RW                         1908K        2 
__TEXT                            19.5M       49 
__UNICODE                          564K        1 
shared memory                        8K        2 
===========                     =======  ======= 
TOTAL                            539.3M      162 
TOTAL, minus reserved VM space   539.2M      162 

Model: MacBookPro14,3, BootROM 428.0.0.0.0, 4 processors, Quad-Core Intel Core i7, 2.8 GHz, 16 GB, SMC 2.45f4
Graphics: kHW_IntelHDGraphics630Item, Intel HD Graphics 630, spdisplays_builtin
Graphics: kHW_AMDRadeonPro555Item, Radeon Pro 555, spdisplays_pcie_device, 2 GB
Memory Module: BANK 0/DIMM0, 8 GB, LPDDR3, 2133 MHz, 0x802C, 0x4D5435324C31473332443450472D30393320
Memory Module: BANK 1/DIMM0, 8 GB, LPDDR3, 2133 MHz, 0x802C, 0x4D5435324C31473332443450472D30393320
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x173), Broadcom BCM43xx 1.0 (7.77.111.1 AirPortDriverBrcmNIC-1615.1)
Bluetooth: Version 7.0.6f7, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB 3.0 Bus
USB Device: Apple T1 Controller
Thunderbolt Bus: MacBook Pro, Apple Inc., 41.4
Thunderbolt Bus: MacBook Pro, Apple Inc., 41.4

**Comment by dkrystki** [comment, marked as answer]
I believe this is a bug in Python 3.7.0. Could you upgrade to 3.7.10 and try again?
Also renaming your project to other name than `reloadium` would be safer.

**Reply by cryptopotluck** [reply]
It's working with python 3.9, but 3.7 isn't compatible at the moment. Cool project and good luck on the programming journey. :)
